staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -199,10 +202,12 @@ func (t *watcherBookmarkTimeBuckets) popExpiredWatchers() [][]*cacheWatcher {
 	// There should be one or two elements in almost all cases
 	expiredWatchers := make([][]*cacheWatcher, 0, 2)
 	for ; t.startBucketID <= currentBucketID; t.startBucketID++ {
+		t.mu.Lock()
```

Comment:
Instead of calling Lock() and unlock() in every iteration of loop (which has non-negligible cost), just call:

t.lock.Lock()
defer t.lock.Unlock()

before the loop